pyccel/ast/builtins.py

```diff
@@ -817,9 +817,14 @@ def print_string(self):
         """ Return a literal string representing the type that
         can be used in a print  statement
         """
-        return LiteralString("<class '{dtype}{precision}'>".format(
-            dtype = str(self.dtype),
-            precision = self.precision*8 if self.precision else ''))
+        prec = self.precision
+        if prec == -1:
+            return LiteralString("<class '{dtype}'>".format(
+                dtype = str(self.dtype)))
+        else:
+            return LiteralString("<class '{dtype}{precision}'>".format(
+                dtype = str(self.dtype),
+                precision = prec*8 if prec else ''))
```

Comment:
Could we not modify the original code just a bit?
```python
        prec = self.precision
        return LiteralString("<class '{dtype}{precision}'>".format(
            dtype = str(self.dtype),
            precision = '' if prec in [None, -1] else prec*8))
```

Reply:
Fixed 88439ee